### PR TITLE
Refactor KubeVirt cloud provider - part 1

### PR DESCRIPTION
**Pull request by mfranczy**
**What this PR does / why we need it**:
One of a few PRs with refactored KubeVirt cloud provider code

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/kubevirt/provider.go [hunk 1]
```diff
@@ -79,6 +78,8 @@ func (k *kubevirt) ValidateCloudSpec(ctx context.Context, spec kubermaticv1.Clou
 		return err
 	}
 
+	// TODO: (mfranczy) this has to be changed
+	// it is wrong to mutate the value of kubeconfig in the validation method
```
Comment (hdurand0710) [inline pkg/provider/cloud/kubevirt/provider.go:83]: +1

**Review by hdurand0710 — APPROVED** [review]

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**